### ISSUE TITLE
flux_sec_keygen make check failures 

### ISSUE THREAD

**Issue by chu11** [flux-framework/flux-core#1028]
Getting these make check failures on the current master ac055a981e668440575ce5c299812fc258b939ed

```
FAIL: test_security.t 24 - flux_sec_keygen fails with EACCES if confdir does not exist
FAIL: test_security.t 25 - flux_sec_keygen (force) fails with EACCES if confdir does not exist
```

**Comment by chu11** [comment]
Some printf-ing saw that the errno was EROFS, "Read-only file system".  I guess it's because the test is against ```/noexist``` and I'm on a system where it's read only.

**Comment by garlick** [comment]
The call path should be `flux_sec_keygen ()`
  -> `checksecdirs  (create=true)` 
  -> `checksecdir (path="/noexist", create=true)`
  -> `mkdir (path="/noexist)`

So I guess mkdir must fail wtih EROFS on a readonly file system and EACCESS otherwise.

Maybe it would be best to check for failure but not require a particular errno, since ENOSPC or EEXIST are not out of the realm of possibility either...

**Comment by chu11** [comment]
That's what I was thinking.   I suppose EPERM is possible too.  EEXIST shouldn't heppen b/c there's a ```lstat()``` check right before the ```mkdir```.

**Comment by garlick** [comment]
Yep, you're right.

**Comment by garlick** [comment]
closed by #1029
